### PR TITLE
Add ui automation

### PR DESCRIPTION
**Pull request by kaisong1990**
## Proposed changes

Add MSAL UI automation and scheduling

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+source "https://rubygems.org"
```
Comment (kaisong1990) [inline Gemfile:1]: This pull request does not update CHANGELOG.md.

Please consider if this change would be noticeable to a partner or user and either update CHANGELOG.md or resolve this conversation.

File: azure_pipelines/automation.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  - name: 'repositoryName' # Name of the repository
+    value: 'AzureAD/microsoft-authentication-library-for-objc'
+
+schedules:
+# daily build will only be triggered if there is any change between the build and last successful build
+- cron: "0 4 * * *"
+  displayName: Daily 8 pm build
+  branches:
+    include:
+    - kaisong1990/msal-ui-automation
```
Comment (oldalton) [inline azure_pipelines/automation.yml:11]: update this to dev?

**Review by oldalton — APPROVED** [review]

File: azure_pipelines/automation.yml (hunk 2 above)
Comment (kaisong1990) [reply azure_pipelines/automation.yml:11]: Yes, this will be updated before merging :)

File: azure_pipelines/automation.yml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      targetType: 'inline'
+      script: |        
+        bundle install
+        echo $(AppleAutomationBase64) > "MSAL/test/automation/conf_encoded.json"
+        base64 --decode MSAL/test/automation/conf_encoded.json > MSAL/test/automation/conf.json
+        ruby TestRubyFile.rb
+        CI=true fastlane install_plugins
+        CI=true fastlane ui_test
+        CI=true bundle exec fastlane ui_test --verbose
+        # Uncomment out the following if want to check test results
```
Comment (ameyapat) [inline azure_pipelines/automation.yml:83]: nit: Add a pipeline variable like debugMode and use if [[ $(debugMode) ]] then
// execute git commands to push results
fi
We can then just flip the variable flag to rerun pipeline with results.

**Review by ameyapat — APPROVED** [review]

File: azure_pipelines/automation.yml (hunk 3 above)
Comment (kaisong1990) [reply azure_pipelines/automation.yml]: Yes this has been updated and published through artifacts

**Pull request merged**